packages/block-library/src/index.js

```diff
@@ -203,15 +202,13 @@ export const registerCoreBlocks = (
  */
 export const __experimentalRegisterExperimentalCoreBlocks =
 	process.env.GUTENBERG_PHASE === 2
-		? ( settings ) => {
-				const { __experimentalEnableFullSiteEditing } = settings;
-
+		? ( isSiteEditor ) => {
```

Comment:
This is not great and this is the main question/uncertainty I have for this PR: When should we enable the FSE blocks? 

IMO, they should be enabled for both FSE and non-FSE themes and in both site editor and post editor. For example you could use a query block in the post editor if you want.

so maybe we can just hide these blocks behind the `process.env.GUTENBERG_PHASE` flag which means they are always enabled on Gutenberg and disabled on Core. The issue is that we didn't think much about their organization in the inserter yet.

Reply:
I'm generally fine registering the site blocks in all editors as they mature. See also #26316. I think several of the theme specific blocks are not quite ready to be fully exposed, though. Also wonder if we'd want a "Theme" category for all these new blocks.

Reply:
Wouldn't we have to ensure that you can't embed Post Content inside the edit post first? There might be other edge cases like can whether it's expected to embed Template Parts inside Post Content. Well, the sooner we enable the sooner we detect potential issues :)

Reply:
Template parts should be restricted to site editing, I don't see a reason to allow them elsewhere outside of causing extra confusion.

Reply:
One thought about this idea:

> so maybe we can just hide these blocks behind the `process.env.GUTENBERG_PHASE` flag which means they are always enabled on Gutenberg and disabled on Core. The issue is that we didn't think much about their organization in the inserter yet.

`process.env.GUTENBERG_PHASE` is compiled away. That means that sites using the package will not be able to opt-in or out of experimental blocks. If that's desirable it would need to remain as a setting.

Reply:
I think I found a better heuristic for the moment: enable these blocks in both post editor and site editor for FSE themes and disable them for regular themes.

The meaning of "FSE themes" might change and once these blocks become more stable, we'll have to revisit this and find better heuristics.